frontend/packages/knative-plugin/src/components/add/event-sources/KafkaSourceSection.tsx

```diff
@@ -1,37 +1,123 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import * as _ from 'lodash';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { InputField, TextColumnField } from '@console/shared';
+import { InputField, SelectInputField } from '@console/shared';
 import { TextInputTypes } from '@patternfly/react-core';
 import KafkaSourceNetSection from './KafkaSourceNetSection';
 import ServiceAccountDropdown from '../../dropdowns/ServiceAccountDropdown';
+import { KafkaModel, KafkaTopicModel } from '../../../models';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getBootstrapServers } from '../../../utils/create-eventsources-utils';
 
 const KafkaSourceSection: React.FC = () => {
-  const { values } = useFormikContext<FormikValues>();
-  const {
-    data: {
-      kafkasource: { bootstrapServers, topics },
-    },
-  } = values;
+  const [bootstrapServers, setBootstrapServers] = React.useState([]);
+  const [bsPlaceholder, setBsPlaceholder] = React.useState('');
+  const [kafkaTopics, setKafkaTopics] = React.useState([]);
+  const [ktPlaceholder, setKtPlaceholder] = React.useState('');
+
+  const kafkaResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(KafkaModel),
+      optional: true,
+    }),
+    [],
+  );
+  const kafkaTopicResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(KafkaTopicModel),
+      optional: true,
+    }),
+    [],
+  );
+  const [kafkasData, kafkasLoaded, kafkasLoadError] = useK8sWatchResource<K8sResourceKind[]>(
+    kafkaResource,
+  );
+  const [kafkaTopicsData, kafkaTopicsLoaded, kafkaTopicsLoadError] = useK8sWatchResource<
+    K8sResourceKind[]
+  >(kafkaTopicResource);
+
+  React.useEffect(() => {
+    let bootstrapServersOptions = [];
+    let placeholder;
+    if (kafkasLoaded && !kafkasLoadError) {
+      bootstrapServersOptions = !_.isEmpty(kafkasData)
+        ? _.map(getBootstrapServers(kafkasData), (bs) => ({
+            value: bs,
+            disabled: false,
+          }))
+        : [
+            {
+              value: 'No Bootstrapservers found',
+              disabled: true,
+            },
+          ];
+      placeholder = 'Add Bootstrapservers';
+    } else {
+      if (kafkasLoadError) {
+        bootstrapServersOptions = [{ value: kafkasLoadError?.message, disabled: true }];
+        placeholder = 'Error loading Bootstrapservers';
+      }
+      bootstrapServersOptions = [{ value: 'Loading Bootstrapservers...', disabled: true }];
+      placeholder = '...';
```

Comment:
The Loading Bootstrapservers messages overrides always the error message here. I think you want wrap it in an else.

Reply:
Oops yeah missed the else here.